test/conformance/tests/sriov_operator.go

```diff
@@ -1109,108 +1115,96 @@ var _ = Describe("[sriov] operator", func() {
 		})
 
 		Context("Resource Injector", func() {
-			// 25834
+
 			AfterEach(func() {
-				cfg := sriovv1.SriovOperatorConfig{}
-				clients.Get(context.TODO(), runtimeclient.ObjectKey{
-					Name:      "default",
-					Namespace: operatorNamespace,
-				}, &cfg)
-				if *cfg.Spec.EnableInjector == false {
-					cfg.Spec.EnableInjector = pointer.BoolPtr(true)
-					clients.Update(context.TODO(), &cfg)
-				}
+				setSriovOperatorSpecFlag(operatorNetworkInjectorFlag, true)
+				setSriovOperatorSpecFlag(operatorWebhookFlag, true)
 			})
 
-			It(" SR-IOV resource injector can be disabled by editing SR-IOV Operator Config	", func() {
+			DescribeTable("SR-IOV Operator Config, disable",
+				func(resourceName string) {
+					var err error
+					setSriovOperatorSpecFlag(resourceName, false)
 
-				networkResourcesInjector := "network-resources-injector"
-				cfg := sriovv1.SriovOperatorConfig{}
-				err := clients.Get(context.TODO(), runtimeclient.ObjectKey{
-					Name:      "default",
-					Namespace: operatorNamespace,
-				}, &cfg)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(*cfg.Spec.EnableInjector).To(BeTrue())
-
-				cfg.Spec.EnableInjector = pointer.BoolPtr(false)
-				err = clients.Update(context.TODO(), &cfg)
-				Expect(err).ToNot(HaveOccurred())
-
-				Eventually(func() bool {
-					_, err := clients.DaemonSets(operatorNamespace).Get(context.Background(), networkResourcesInjector, metav1.GetOptions{})
-					if k8serrors.IsNotFound(err) {
-						return true
-					}
-					Expect(err).ToNot(HaveOccurred())
-					return false
-				}, 3*time.Minute, 20*time.Second).Should(BeTrue())
+					Eventually(func() bool {
+						_, err := clients.DaemonSets(operatorNamespace).Get(context.Background(), resourceName, metav1.GetOptions{})
+						if k8serrors.IsNotFound(err) {
+							return true
+						}
+						Expect(err).ToNot(HaveOccurred())
+						return false
+					}, 3*time.Minute, 20*time.Second).Should(BeTrue())
 
-				Eventually(func() bool {
+					Eventually(func() bool {
 
-					podsList, err := clients.Pods(operatorNamespace).List(context.Background(), metav1.ListOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					for _, pod := range podsList.Items {
-						if strings.Contains(pod.ObjectMeta.Name, networkResourcesInjector) {
-							return false
+						podsList, err := clients.Pods(operatorNamespace).List(context.Background(), metav1.ListOptions{})
```

Comment:
just use a label selector here and wait for the list to be empty

Reply:
Fixed